### PR TITLE
[Enhancement] Pokemon generation number tooltip

### PR DESCRIPTION
**Pull request by EmberCM**
## What are the changes?
Adds a tooltip to show the generation of the Pokemon

## Why am I doing these changes?
I don't wanna have to check a pokedex for every Pokemon during a challenge run

## What did change?
Added a tooltip when you hover over the enemy Pokemon's name to show their generation number

### Screenshots/Videos
tooltip
mobile
setting

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

tests

### CODE REVIEW

**Comment by OrangeRed** [comment]
Instead of numbers can you use roman numerals like in the starter select menu

**Comment by EmberCM** [comment]
> Instead of numbers can you use roman numerals like in the starter select menu

This will be aligned with `locales\starter-select-ui-handler.ts`
rn

**Comment by torranx** [comment]
If this is entirely a tooltip _on hover_, I dont see the need to create a dedicated option for this. The "option" is essentially when the user decides to hover the name.

**Comment by ReneGV** [comment]
Given that the change is meant to help challenge runs I am not sure if there is overlap with
- https://github.com/pagefaultgames/pokerogue/pull/2628

File: src/battle-scene.ts [hunk 1]
```diff
@@ -173,6 +173,14 @@ export default class BattleScene extends SceneBase {
...
   public typeHints: boolean = false;
 
+  /**
+   * Determines when to show the generation tooltip
+   * - 0 = Never
+   * - 1 = During Challenges
+   * - 2 = Always
+   */
+  public generationTooltip: number = 0;
+
```
Comment (OrangeRed) [inline src/battle-scene.ts:183]: you want to fixed values like this use the literal type `0 | 1 | 2` instead of number or just use the enum as the typedef

**Review by OrangeRed — CHANGES_REQUESTED** [review]

File: src/battle-scene.ts (hunk 1 above)
Comment (EmberCM) [reply src/battle-scene.ts]: Instance variable no longer exists

**Comment by EmberCM** [comment]
> If this is entirely a tooltip _on hover_, I dont see the need to create a dedicated option for this. The "option" is essentially when the user decides to hover the name.

I added the option in case people only wanted to only have it show up during challenges, but you make a good point.

**Comment by torranx** [comment]
> > If this is entirely a tooltip _on hover_, I dont see the need to create a dedicated option for this. The "option" is essentially when the user decides to hover the name.
> 
> 
> 
> I added the option in case people only wanted to only have it show up during challenges, but you make a good point.

My point stands that if the user doesnt want it to show up, then avoid hovering it, also its not as if the game is played using mouse so there is always the intention if the user decides to hover it. This is only my opinion tho. Its up for the devs to decide

**Comment by marcodola** [comment]
I worked on [ [Feature]Highlight usable pokemon in green during a challenge #2628 ](https://github.com/pagefaultgames/pokerogue/pull/2628) but I didn't consider the option of the tooltip and it seems great. 
I was wondering if it's possible to merge the two ideas by showing a message like "This pokemon is suitable for the current challenge" or something similar

File: src/locales/fr/battle-info.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleInfo: SimpleTranslationEntries = {
+  "generation": "Generation {{generation}}",
```
Comment (Adri1) [inline src/locales/fr/battle-info.ts:4]:
```suggestion
  "generation": "Génération {{generation}}",
```

File: src/locales/de/battle-info.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleInfo: SimpleTranslationEntries = {
+  "generation": "Generation {{generation}}",
```
Comment (CodeTappert) [inline src/locales/de/battle-info.ts:4]:
```suggestion
  "generation": "{{generation}}. Generation",
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/pt_BR/battle-info.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleInfo: SimpleTranslationEntries = {
+  "generation": "Generation {{generation}}",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle-info.ts:4]:
```suggestion
  "generation": "Geração {{generation}}",
```

File: src/locales/zh_TW/battle-info.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleInfo: SimpleTranslationEntries = {
+  "generation": "Generation {{generation}}",
```
Comment (RimKnight) [inline src/locales/zh_TW/battle-info.ts:4]:
```suggestion
  "generation": "第{{generation}}世代",
```

File: src/locales/zh_CN/battle-info.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleInfo: SimpleTranslationEntries = {
+  "generation": "Generation {{generation}}",
```
Comment (RimKnight) [inline src/locales/zh_CN/battle-info.ts:4]:
```suggestion
  "generation": "第{{generation}}世代",
```

File: src/locales/ko/battle-info.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleInfo: SimpleTranslationEntries = {
+  "generation": "Generation {{generation}}",
```
Comment (EnochG1) [inline src/locales/ko/battle-info.ts:4]:
```suggestion
  "generation": "{{generation}}세대",
```

**Review by EnochG1 — COMMENTED** [review]
Korean translation

**Comment by NicusPulcis** [comment]
@EmberCM i've added the italian translation

**Review by CodeTappert — APPROVED** [review]

**Pull request merged**